events/src/camera.rs

```diff
@@ -23,31 +33,57 @@ pub struct Camera {
     /// Model
     pub model: Option<String>,
 
-    /// Connection type:
-    /// - `local` for USB and CSI cameras
-    /// - `remote` for IP cameras
-    pub conn_type: Option<String>,
+    /// Camera's physical interface
+    pub interface: LocalCameraInterface,
 
-    /// Physical interface: `usb`, `csi`, `ethernet`
-    pub interface: Option<String>,
+    /// List of camera capabilities
+    pub capabilities: Vec<Capability>,
+}
 
-    /// URI to access video:
-    /// - local file name for `local` cameras (for example `/dev/video0`)
-    /// - RTSP URI for `remote` cameras (for example `rtsp://192.168.1.2/my_stream`)
-    pub uri: Option<String>,
+/// `RemoteCamera` type represents IP cameras.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct RemoteCamera {
+    /// ONVIF device management service URI used to access this camera,
+    /// for example `http://192.168.0.42/device`
+    pub uri: Url,
+
+    /// Status
+    pub status: Status,
+
+    /// Name
+    pub name: Option<String>,
 
-    /// Local IP address for IP cameras
+    /// Manufacturer
+    pub manufacturer: Option<String>,
+
+    /// Model
+    pub model: Option<String>,
+
+    /// Local IP address
     pub ip_local: Option<String>,
 
-    /// MAC address for IP cameras
+    /// MAC address
     pub mac_address: Option<String>,
 
-    /// List of camera capabilities
-    pub capabilities: Vec<Capability>,
+    /// List of camera streams
+    pub streams: Vec<Stream>,
+}
+
+/// IP camera stream.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct Stream {
+    /// URL to access this stream, for example `rtsp://192.168.0.42:554/hd_stream`
+    pub rtsp_uri: Url,
+
+    /// Stream name
+    pub name: Option<String>,
```

Comment:
name is optional?

Reply:
Hm, just rechecked and ONVIF stores stream name as required, so should we. Good catch!